Models/Connection/ConnectionInfo.js

```diff
@@ -51,6 +51,18 @@ export default class ConnectionInfo {
   loadPageAsync(pageName, rawCommand, pageSize, domainId, cancellationToken) {
     throw new Error("ConnectionInfo.loadPageAsync() method not implemented.");
   }
+/**
+ * Insert a document into a MongoDB collection.
+ * @param {string} sourceName
+ * @param {string} connectionName 
+ * @param {string} databaseName 
+ * @param {string} collectionName 
+ * @param {Object} data 
+ * @returns {Promise<Object>} 
+ */
+async insertAsync(sourceName, connectionName, databaseName, collectionName, data) {
+  throw new Error("ConnectionInfo.insertAsync() method not implemented.");
+}
```

Comment:
Why should this method be declared in the base class?
Declare internally in the related class